Source/CryptRaider/Actor/Trigger/Dialog/DialogTrigger.cpp

```diff
@@ -2,19 +2,65 @@
 
 #include "DialogTrigger.h"
 
+#include "CryptRaider/Player/BasePlayerController.h"
+
+void ADialogTrigger::OnConstruction(const FTransform& Transform)
+{
+	Super::OnConstruction(Transform);
+
+	if (DialogStrategyClass)
+	{
+		if (!DialogStrategy || DialogStrategy->GetClass() != DialogStrategyClass)
+		{
+			if (DialogStrategy)
+			{
+				DialogStrategy->ConditionalBeginDestroy();
```

Comment:
Let's check if we can use BeginDestroy instead

ConditionalBeginDestroy looks confusing

https://dev.epicgames.com/documentation/en-us/unreal-engine/unreal-engine-actor-lifecycle

Reply:
Otherwise it throws exception:

```
LoginId:7415b82546caa5c86b3a4eb61499ee0b
EpicAccountId:59f4744e7c4c4f01afd713634939bba2

Fatal error: [File:D:\build\++UE5\Sync\Engine\Source\Runtime\CoreUObject\Private\UObject\Obj.cpp] [Line: 942] Trying to call UObject::BeginDestroy from outside of UObject::ConditionalBeginDestroy on object State Strategy. Please fix up the calling code.

UnrealEditor_CoreUObject
UnrealEditor_CryptRaider_patch_0!ADialogTrigger::OnConstruction() [D:\epicProj\CryptRaiderGame\Source\CryptRaider\Actor\Trigger\Dialog\DialogTrigger.cpp:31]
UnrealEditor_Engine
UnrealEditor_Engine
UnrealEditor_Engine
UnrealEditor_CoreUObject
UnrealEditor_PropertyEditor
UnrealEditor_PropertyEditor
UnrealEditor_PropertyEditor
UnrealEditor_PropertyEditor
UnrealEditor_PropertyEditor
UnrealEditor_PropertyEditor
UnrealEditor_PropertyEditor
UnrealEditor_PropertyEditor
UnrealEditor_PropertyEditor
UnrealEditor_ClassViewer
UnrealEditor_ClassViewer
UnrealEditor_ClassViewer
UnrealEditor_ClassViewer
UnrealEditor_ClassViewer
UnrealEditor_ClassViewer
UnrealEditor_Slate
UnrealEditor_Slate
UnrealEditor_Slate
UnrealEditor_Slate
UnrealEditor_ApplicationCore
UnrealEditor_ApplicationCore
UnrealEditor_ApplicationCore
UnrealEditor_ApplicationCore
user32
user32
UnrealEditor_ApplicationCore
UnrealEditor
UnrealEditor
UnrealEditor
UnrealEditor
UnrealEditor
UnrealEditor
kernel32
ntdll
```